### PR TITLE
Calendroid: Fix notification scheduler

### LINKED ISSUE #4244: Implement/Fix Android notification service
It has to actually use the `Multivac.Notification.ScheduledTime` and notify the user only at the expected time.

This issue should take care of scheduling the notifications only. Re-scheduling notifications after the device boots will have it's own issue.

Hints: Use the `AlarmManager` to schedule a pending intent to run at the specified time and create a `BroadcastReceiver` to actually send the notifications (call `NotificationManager.Notify`)

### PR DESCRIPTION
**Pull request by daividssilverio**
## What's this?
This fixes the android `NotificationServiceAndroid.cs` to actually schedule notifications using the `Multivac.Notification:ScheduledTime`.

### Relationships
Closes #4244 

## Why do we want this?
To show calendar event notifications based on smart alert settings.

## How is it done?
Instead of firing the notifications on the spot, it uses the `AlarmManager` to send an intent that will be handled by the new broadcast listener that's responsible for actually firing the notifications.

### Why not another way?
This is easy enough.

### Side effects
None, this service is used by the Calendar only.

## Review hints
You can use this snippet to see it working.
It doesn't persist schedules through a device restart, that's gonna be done at #4245.

## :squid: Permissions
🦑

### CODE REVIEW

File: Toggl.Giskard/Services/NotificationServiceAndroid.cs [hunk 1]
```diff
@@ -26,15 +30,16 @@ public IObservable<Unit> Schedule(IImmutableList<Multivac.Notification> notifica
 
                     notifications.Select(notification =>
```
Comment (heytherewill) [inline Toggl.Giskard/Services/NotificationServiceAndroid.cs]: Do you even need to select and then `foreach` at this point? This is not doing any good: it just adds makes you traverse the collection twice and adds an indirection.

Might as well fix this with a simple:
```
foreach(var notification in notifications)
{
    var notificationId = notification.Id.GetHashCode();
    var androidNofitication = notificationsBuilder.SetAutoCancel(true)
                                .SetContentIntent(pendingIntent)
                                .SetContentTitle(notification.Title)
                                .SetSmallIcon(Resource.Drawable.ic_icon_running)
                                .SetContentText(notification.Description)
                                .Build();

    scheduleNotification(notificationId, androidNotification, notification.ScheduledTime);
}
```

**Comment by daividssilverio** [comment]
Oh, the code snippet:
```
var context = Application.Context;
            var notificationIntent = context.PackageManager.GetLaunchIntentForPackage(context.PackageName);
            notificationIntent.SetPackage(null);
            var pendingIntent = PendingIntent.GetActivity(context, 0, notificationIntent, 0);
            var notificationManager = NotificationManager.FromContext(context);
            var notificationsBuilder = context.CreateNotificationBuilderWithDefaultChannel(notificationManager);

            NotificationServiceAndroid.scheduleNotification(
                new Random().Next(),
                notificationsBuilder.SetAutoCancel(true)
                    .SetContentIntent(pendingIntent)
                    .SetContentTitle("Nice Titlte")
                    .SetSmallIcon(Resource.Drawable.ic_icon_running)
                    .SetContentText("Nice Content!")
                    .Build(),
                DateTimeOffset.Now + TimeSpan.FromMinutes(2)
            );
```

File: Toggl.Giskard/Services/NotificationServiceAndroid.cs (hunk 1 above)
Comment (daividssilverio) [reply Toggl.Giskard/Services/NotificationServiceAndroid.cs]: it's still a foreach, right?

Comment (heytherewill) [reply Toggl.Giskard/Services/NotificationServiceAndroid.cs]: It is

**Pull request merged**